discover/iterator.go

```diff
@@ -35,3 +37,56 @@ func generateIterators(list []string) []*Iterator {
 
 	return result
 }
+
+func findCommonDirectory(effxFileLocations []string) string {
+	matchedEffxFiles := generateIterators(effxFileLocations)
+	prefixString := ""
+
+	for len(matchedEffxFiles) > 0 {
+		count := make(map[string]int)
```

Comment:
let's call this `charCounts`. I think it better communicates whats happening in the loop.